nntrainer/utils/base_properties.h

```diff
@@ -316,6 +322,39 @@ template <typename Tag, typename DataType> struct str_converter {
   static DataType from_string(const std::string &value);
 };
 
+/**
+ * @brief str converter which serializes a pointer and returns back to a ptr
+ *
+ * @tparam DataType pointer type
+ */
+template <typename DataType> struct str_converter<ptr_prop_tag, DataType> {
+
+  /**
+   * @brief convert underlying value to string
+   *
+   * @param value value to convert to string
+   * @return std::string string
+   */
+  static std::string to_string(const DataType &value) {
+    std::ostringstream ss;
+    ss << value;
+    return ss.str();
+  }
+
+  /**
+   * @brief convert string to underlying value
+   *
+   * @param value value to convert to string
+   * @return DataType converted type
+   */
+  static DataType from_string(const std::string &value) {
+    std::stringstream ss(value);
```

Comment:
I wonder if this is just for consistency with others or actual use cases? I think if the user wants to set the user_data from string, then they might set the string from outside of function and hand it over as a pointer.

Reply:
I am thinking this is strictly internal use only and we should not allow user to set pointer.
The reasoning here is that, properties are designed to be saved to a file while saving an address will have no meaning because of address space layout randomization.

For the user_data, user_data can be set at runtime with below two functions and setting from ini won't be allowed.

1. `ml_train_set_property_for_mode(dataset, TRAIN, "user_data", &user_data, nullptr);` // this is more for the backward compatibility
2. `ml_train_dataset_add_generator(dataset, TRAIN, user_func, &user_data);`